tests/slepos/register_images.pm

```diff
@@ -17,13 +17,53 @@ use testapi;
 use utils;
 use lockapi;
 
+my $img_suffix = 'tar.bz2';
+
 sub run() {
-    assert_script_run "registerImages --ldap --move --include-boot /var/lib/SLEPOS/system/images/minimal-3.4.0/";
-    assert_script_run "registerImages --gzip --ldap /var/lib/SLEPOS/system/images/graphical-3.4.0";
 
-    assert_script_run "curl " . autoinst_url . "/data/slepos/xorg.conf > /srv/SLEPOS/config/xorg.conf";
-    assert_script_run
+    my $kiwi_images_ref    = get_var_array('IMAGE_KIWI');
+    my $creator_images_ref = get_var_array('IMAGE_CREATOR');
+    my $dwl_images_idx_ref = get_var_array('IMAGE_DOWNLOAD');
+    my @dwl_images;
+    my @kiwi_images;
+    my @creator_images;
+    my $pxe_done = 0;
+    foreach my $idx (@$dwl_images_idx_ref) {
+        my $target = get_var("ASSET_$idx");
+        $target =~ s/^\d+-//;                #remove if private assets reference
+        $target =~ s/\Q.$img_suffix\E$//;    #remove suffix
+        @dwl_images = (@dwl_images, $target);
```

Comment:
There seems to be some code duplication between this and download_images.pm. Maybe you could move the image name processing to a library.